opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -2206,159 +2207,122 @@ void CpGrid::addLgrsUpdateLeafView(const std::vector<std::array<int,3>>& cells_p
+                        const auto& equivPoint = cpgrid::Entity<3>(*( (*current_data_)[bornLevel_bornIdx[0]]), bornLevel_bornIdx[1], true);
+                        localToGlobal_points_per_level[level-1][point.index()] =
+                            current_data_->front()->global_id_set_->id( equivPoint );
+                    }
+                    else {
+                        // Assign a global ID to points that do not coincide with any corners from level zero.
+                        // This ID may be overwritten later when parallel indices and level cell index sets
+                        // are defined for all level grids, particularly in cases where LGRs may be distributed
+                        // across processes.
+                        localToGlobal_points_per_level[level-1][point.index()] = min_globalId_point_in_proc;
+                        ++min_globalId_point_in_proc;
                     }
                 }
             }
-            localToGlobal_owned_cells_per_level[level-1].shrink_to_fit(); // Not needed in this special case without overlap refined cells
-            localToGlobal_owned_points_per_level[level-1].shrink_to_fit(); // Not needed in this special case without overlap refined points
-
-            // Currently, only fully interior LGRs are supported. Therefore, global_id_set_
-            // is defined only in the process that owns at least one LGR.
-            if(lgrs_with_at_least_one_active_cell[level-1]>0) {
-                (*current_data_)[level]->global_id_set_->swap(localToGlobal_owned_cells_per_level[level-1],
-                                                              localToGlobal_owned_faces_per_level[level-1],
-                                                              localToGlobal_owned_points_per_level[level-1]);
+            // For the general case where the LGRs might be also distributed, a communication step is needed to assign global ids
+            // for overlap cells and points.
+
+            // Global id set for each (refined) level grid.
+            if(lgr_with_at_least_one_active_cell[level-1]>0) {
+                (*current_data_)[level]->global_id_set_->swap(localToGlobal_cells_per_level[level-1],
+                                                              localToGlobal_faces_per_level[level-1],
+                                                              localToGlobal_points_per_level[level-1]);
             }
         } // end-for-loop-level
-        assert(globalIdCell <= max_globalId_levelZero + global_refined_cell_count +1);
-        assert(globalIdPoint <= max_globalId_levelZero + global_refined_cell_count + almost_global_refined_point_count+1);
-        
 
         ////////////////////////////////
 
         // Global id for the cells in leaf grid view
-        std::vector<int> leafCellIds(current_data_->back()->size(0));
+        std::vector<int> leafCellIds(current_data_->back()->size(0), 0);
```

Comment:
I probably should have commented on all these. Here is another one with an access parameter that is the default value, anyway. Maybe remove it, too.

Reply:
I could have also checked, removed!